File: beecrowd/python/grafos/ex2300.py
def bfs(o, m, v):
    # o = origem
    # m = matriz
    # v = vertices
    visitados = [0] * v
    visitados[o] = 1
    fila = [o]
    
    while len(fila) != 0:
        v_atual = fila[0]
        del fila[0]
        
        for vz in range(v):
            if visitados[vz] == 0 and m[v_atual][vz] == 1:
                visitados[vz] = 1
                fila.append(vz)
                
    return visitados

ct = 0
while True:
    ct += 1
    e, l = map(int, input().split())
    # e = estacoes
    # l = linhas
    if e == l == 0: break
    
    m = [[0] * e for _ in range(e)]
    # m = matriz
        
    for _ in range(l):
        x, y = map(int, input().split())
        m[x-1][y-1] = 1
        m[y-1][x-1] = 1

    if bfs(0, m, e) == [1] * e:
        falha = False
    else:
        falha = True
    
    print(f'Teste {ct}')
    
    if falha:
        print('falha')
    else:
        print('normal')
    
    print()

File: beecrowd/c++/grafos/ex2300.cpp
#include <bits/stdc++.h> 
using namespace std;

int matriz[100][100];
int visitados[100];
int v;

void bfs(int origem){
    queue<int> fila;
    fila.push(origem);
    visitados[origem] = 1;

    while(!fila.empty()){
        int v_atual = fila.front();
        fila.pop();

        for (int vz=0; vz <= v; vz++){
            if (visitados[vz] == 0 and matriz[v_atual][vz] == 1){
                fila.push(vz);
                visitados[vz] = 1;
            }
        }
    }
}

int main(){
    int ct = 0;
    int e, l;
    cin>>e>>l;
    do{
        ct++;
        memset(matriz, 0, sizeof(matriz)); 
        memset(visitados, 0, sizeof(visitados));
        v = e;

        for(int i=0; i<l; i++){
            int x, y;
            cin>>x>>y;
            matriz[x-1][y-1] = 1;
            matriz[y-1][x-1] = 1;
        }

        bfs(0);
        bool falha = false;
        for(int i=0; i<v; i++){
            if(visitados[i] == 0) 
                falha = true;
        }

        cout<<"Teste "<<ct<<endl;
        if(falha)
            cout<<"falha"<<endl;
        else   
            cout<<"normal"<<endl;

        cout<<endl;
        cin>>e>>l;

    } while(e != 0 && l != 0);

}